### DISCUSSION TITLE
[BUG] Auto Wild Surge after Tides of Chaos and Tides of Chaos Recharge not working

### DISCUSSION THREAD

**Discussion by wickermoon** [johnnolan/wild-magic-surge-5e#159] [General]
**Describe the bug**
When casting a spell after having used up my tides of chaos feature, the auto recharge and auto surge do not trigger.

**What version of FoundryVTT are you using?**
Foundry VTT v9 Build 269
DnD5e - Fifth Edition System 1.6.3

**To Reproduce**
Set up two feats: "Wild Magic Surge" and "Tides of Chaos" and a roll table "Wild Magic Surge Table"
Activate "Tides of Chaos auto recharge" and "Auto surge on spell use after Tides of Chaos has been used"
Activate "Auto Roll Table" and update Roll Table name to "Wild Magic Surge Table"
Use "Tides of Chaos", so its charge is depleted
Cast "Alarm", which triggers a Wild Magic Surge Check

**Expected behavior**
Instead of a wild magic surge check, an automatic roll on the wild magic surge table should trigger and the tides of chaos feature should be recharged.

**Screenshots**
![image](https://user-images.githubusercontent.com/15907070/182566660-bb84aa69-7f95-4449-88f2-cb209ba2838f.png)
![image](https://user-images.githubusercontent.com/15907070/182568043-292b64b4-7685-4ca8-a310-68c13a792f27.png)
![image](https://user-images.githubusercontent.com/15907070/182568100-995d7afe-ddc9-4127-a7dc-00216b1f5e8e.png)

**Comment by johnnolan** [comment]
Hi, thanks for the report, are you able to post in here your character? 

If you right click the actor in the side menu and choose export data, then copy and paste the data here. Then I can see what might be missing and update the readme or code

**Comment by wickermoon** [comment]
[fvtt-Actor-lorry.json.txt](https://github.com/johnnolan/wild-magic-surge-5e/files/9250958/fvtt-Actor-lorry.json.txt)

There you go

**Comment by johnnolan** [comment]
Okay, it looks like it requires the resource consumption on Tides of Chaos to be set in order to work. I have updated the actor and attached the working version.

Will update the readme to be more helpful for setting an actor up and also in V10 see if there is a way to better handle usage than having to do this extra step.

Let me know if that works for you


![image](https://user-images.githubusercontent.com/5390820/182602012-d03f64f3-0162-4901-a298-b3496d4ab272.png)
![image](https://user-images.githubusercontent.com/5390820/182602070-57b0c8a8-6f40-4957-9875-02417d2285e8.png)
[fvtt-Actor-lorry(1).json.txt](https://github.com/johnnolan/wild-magic-surge-5e/files/9251035/fvtt-Actor-lorry.1.json.txt)

**Comment by wickermoon** [comment]
That fixed both problems. :)
